src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,24 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $cell-gap;
+
+  width: width($cell-size, $cell-gap);
+  padding: 10px;
+
+  @each $name, $day in $week {
+    &--start-day--#{$name} {
+      & > :first-child {
+        margin-left: dayGap($cell-size, $cell-gap, $day);
+      }
+    }
+  }
```

Comment:
Here you are breaking BEM rules 'cuz your `calendar `block modifies a separate day `block`.
To solve this problem you could use mixes, or just make all of your days the calendar's elements (`calendar__day`) what is better